levels/puzzles/TVPuzzle.gd

```diff
@@ -0,0 +1,164 @@
+"""
+Handles logic for the first puzzle
+Puzzle state is represented by two coordinates:
+	Position of the antenna on the X axis
+	Selected channel on the Y axis
+"""
+extends Node2D
+signal puzzle_quit
+signal puzzle_won
+
+const MIN_VALUE = 0
```

Comment:
Each puzzle should have at least 2 state: solved and not solved. They will affect to how the puzzle appeared. But I'm not sure where is best place to put this logic. I though about abstract class for Puzzle with a solved sprite. The abstract class will handle solved case by display solved sprite, disable all interaction with puzzle.
Puzzle which have solved partially concept is more complex and should be considered later

Reply:
Like you said before, I will have a state for the TV: fixed (bool). When TV is fixed, puzzle will simply not be triggered.